99-monitor.rules

```diff
@@ -0,0 +1 @@
+ACTION=="change", SUBSYSTEM=="drm", ENV{DISPLAY}=":0", ENV{XAUTHORITY}="/home/{username}/.Xauthority", RUN+="/usr/bin/xlayoutdisplay"
```

Comment:
This works nicely. I've tweaked it a bit, so that we can pick up the user's ~/.xlayoutdisplay and log the output:

```
ACTION=="change", SUBSYSTEM=="drm", GOTO="xlayoutdisplay"

LABEL="xlayoutdisplay"

ENV{HOME}="/home/alex"
ENV{DISPLAY}=":0"
ENV{XAUTHORITY}="$env{HOME}/.Xauthority"

RUN+="/bin/sh -c 'xlayoutdisplay >> /tmp/xlayoutdisplay.udev.log 2>&1'"
```

Can you please test the above?

Reply:
We could also add `ENV{XDG_CONFIG_HOME}="$env{HOME}/.config"` if the user has their config file in an alternate location, but that can just go in the readme.

Reply:
> This works nicely. I've tweaked it a bit, so that we can pick up the user's ~/.xlayoutdisplay and log the output:
> 
> ```
> ACTION=="change", SUBSYSTEM=="drm", GOTO="xlayoutdisplay"
> 
> LABEL="xlayoutdisplay"
> 
> ENV{HOME}="/home/alex"
> ENV{DISPLAY}=":0"
> ENV{XAUTHORITY}="$env{HOME}/.Xauthority"
> 
> RUN+="/bin/sh -c 'xlayoutdisplay >> /tmp/xlayoutdisplay.udev.log 2>&1'"
> ```
> 
> Can you please test the above?

Using the tweaked rule, i could not boot my reboot my system. According to `journalctl` it seems that the rule itself, or the process triggered by the rule, would be stuck in a loop and therefore blocking other processes, like mounting partitions. 

It seemed for me, that this was rather a problem with the formatting of the rule, than the rule itself. After removing the newlines and make it a one-liner (seperated by commas), the system booted again and the rule itself executed just fine.

I will check again if we could use your tweaked rule, as it's way cleaner than having a really long one-liner.

Reply:
Thank you; my testing was not thorough.

Merging now.

I'll keep testing...

Reply:
> Using the tweaked rule, i could not boot my reboot my system. According to `journalctl` it seems that the rule itself, or the process triggered by the rule, would be stuck in a loop and therefore blocking other processes, like mounting partitions.

Unfortunately I am seeing this issue with the simple rule. It seems we are hitting https://wiki.archlinux.org/title/Udev#X_programs_in_RUN_rules_hang_when_no_X_server_is_present

We need some sort of means of detecting whether X is actually running. It could be something in the rule or something in the code...

Reply:
I've added detection for an `Xorg` process to the executable: https://github.com/alex-courtis/xlayoutdisplay/tree/19-detect-xorg-running

Can you please test that branch through the reboot cycle

Reply:
Tested and works with your tweaked rule.
Log is also filled as expected.

Thank you for the quick fix :)

Reply:
Thanks for testing. I'll let things soak for a few days then raise a PR. I also need to test the complex rule.